gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleModuleProjectProvider.java

```diff
@@ -0,0 +1,210 @@
+import org.eclipse.buildship.core.util.gradle.GradleDistributionWrapper;
+import org.eclipse.buildship.core.util.progress.AsyncHandler;
+import org.eclipse.buildship.core.util.progress.DelegatingProgressListener;
+import org.eclipse.buildship.core.workspace.SynchronizeGradleProjectJob;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.sapphire.ElementList;
+import org.eclipse.sapphire.platform.PathBridge;
+import org.gradle.tooling.CancellationToken;
+import org.gradle.tooling.ProgressListener;
+
+/**
+ * @author Gregory Amerson
+ * @author Terry Jia
+ * @author Andy Wu
+ */
+public class LiferayGradleModuleProjectProvider extends LiferayModuleProjectProvider
+{
+
+    public LiferayGradleModuleProjectProvider()
+    {
+        super( new Class<?>[] { IProject.class } );
+    }
+
+    @Override
+    public synchronized ILiferayProject provide( Object adaptable )
+    {
+        ILiferayProject retval = null;
+
+        if( adaptable instanceof IProject )
+        {
+            final IProject project = (IProject) adaptable;
+
+            try
+            {
+                if( LiferayNature.hasNature( project ) && GradleProjectNature.INSTANCE.isPresentOn( project ) )
+                {
+                    return new LiferayGradleProject( project );
+                }
+            }
+            catch( Exception e )
+            {
+                // ignore errors
+            }
+        }
+
+        return retval;
+    }
+
+    private static final class NewGradleProjectInitializer implements AsyncHandler
+    {
+
+        private final FixedRequestAttributes fixedAttributes;
+        private final Optional<List<ProgressListener>> listeners;
+
+        private NewGradleProjectInitializer( ProjectImportConfiguration configuration )
+        {
+            this.fixedAttributes = configuration.toFixedAttributes();
+            this.listeners = Optional.absent();
+        }
+
+        private NewGradleProjectInitializer( ProjectImportConfiguration configuration, List<ProgressListener> listeners )
+        {
+            this.fixedAttributes = configuration.toFixedAttributes();
+            this.listeners = Optional.of( (List<ProgressListener>) ImmutableList.copyOf( listeners ) );
+        }
+
+        @Override
+        public void run( IProgressMonitor monitor, CancellationToken token )
+        {
+            monitor.beginTask( "Init Gradle project", IProgressMonitor.UNKNOWN );
+            try
+            {
+                File projectDir = this.fixedAttributes.getProjectDir().getAbsoluteFile();
+                if( projectDir.exists() )
+                {
+                    // prepare the request
+                    final ImmutableList<String> GRADLE_INIT_TASK_CMD_LINE =
```

Comment:
WHere is this code from? Is it a copy of what the new gradle project import wizard does?  I wouldn't think that an "import" should need to call "init"--type "java-library" because that is the gradle command for creating a new project.  Have you researched what  the "import" gradle wizard does?  That is the code that we should mirror.

Because this is the process:

1. we use blade to create the base project from template
2. then we use gradle import wizard style code to import that newly created project.